r1.sh

```diff
@@ -1,5 +1,15 @@
 #!/bin/bash
 
+# Current working directory, but slash escaped. Useful for sed search/replace
+# strings.
+ESCAPED_PWD=`echo "$PWD" | sed -e "s@\/@\\\\\/@g"`
+
+# Python that we will install in a venv.
+SEDSTR_BINPYTHON=`echo s@/usr\/bin\/python@${ESCAPED_PWD}\/venv\/bin\/python@g`
+SEDSTR_ENVPYTHON=`echo s@/usr\/bin\/env python@${ESCAPED_PWD}\/venv\/bin\/python@g`
+VENV_PYTHON=$PWD/venv/bin/python
+VENV_PIP=$PWD/venv/bin/pip3
```

Comment:
python and pip3 should always point to the exec/link in venv, will this be necessary?

Reply:
That was an assumption that somehow didn't hold true. The reason is that despite activating the virtualenv (which changes the path), if a user uses `sudo` it can mess with the path again, which can mean a different python is used. By forcing all references to refer to the virtualenv, we ensure it uses that python specifically.

Reply:
But running the script with sudo should setup the env for root; after all, the entire script is run either in sudo or not, and the script handles venv creation and activate

Reply:
FWIW I didn't run the whole script in `sudo` and many calls to sudo turned out to be superfluous (e.g. mtkbootcmd.py didn't need sudo on my machine to work).